src/lucky/params.cr

```diff
@@ -70,10 +107,50 @@ class Lucky::Params
     @_form_params ||= HTTP::Params.parse(body)
   end
 
+  @_multipart_params : Hash(String, String)?
+
+  private def multipart_params : Hash(String, String)
+    @_multipart_params ||= parse_multipart_request.first
+  end
+
+  @_multipart_files : Hash(String, Tempfile)?
+
+  private def multipart_files : Hash(String, Tempfile)
+    @_multipart_files ||= parse_multipart_request.last
+  end
+
+  @_parsed_multipart_request : Tuple(Hash(String, String), Hash(String, Tempfile))?
+
+  private def parse_multipart_request
+    @_parsed_multipart_request ||= begin
+      multipart_params = {} of String => String
+      multipart_files = {} of String => Tempfile
+      body_io = IO::Memory.new(body)
+      boundary =
+        HTTP::Multipart.parse_boundary(request.headers["Content-Type"]).to_s
+      HTTP::FormData.parse(body_io, boundary.to_s) do |part|
+        case part.headers
+        when .includes_word?("Content-Disposition", "filename")
+          part_file = Tempfile.open(part.name) do |tempfile|
+            IO.copy(part.body, tempfile)
+          end
+          multipart_files[part.name] = part_file
```

Comment:
@paulcsmith I've been thinking about handling metadata. I could, at this point, extract all of the available metadata to `model:#{file_param_name}_content_type` etc. params. This would then automatically populate the model appropriately.